### DISCUSSION TITLE
iFi hip-dac with mpd on Android 7.1 (Lineage 14.1) tablet

### DISCUSSION THREAD

**Discussion by tHubig** [MusicPlayerDaemon/MPD#1815] [Q&A]
How do you specify the audio device in the audio_output{} section of mpd.conf if you want to use an external USB DAC?

My tablet, LG Gpad 8 v521, has a built-in USB (A) host. I plugged in the hip-dac and it works with Unpopular Music player,  Opus 1, but those players will not send DSF or FLAC to the hip-dac. In fact, they won't play them at all even on the internal built-in DAC. 

So I tried mpd + the M.A.L.P. player but it does not see the DAC at all. MALP plays fine to the built-in audio, whatever piece of crap it is, but even 128kbps  mp3s sound so MUCH better using the hip-dac. (mpd is able to play 88.2 kHz/ 24-bit flacs but it is a complete waste using the tablet's internal rubbish DAC.)

The tablet is rooted and running what I think was the last version of Lineage for it.
I installed mpd from F-droid. 

I've configured mpd to use this DAC to natively play DSD and flacs (above 44.1/16) on several Linux PCs but I'm completely lost in Android becuase it doesn't use ALSA but some strange other thing....

**Comment by MaxKellermann** [comment]
What Android API is available for address this DAC?
